pmd-core/src/main/java/net/sourceforge/pmd/lang/document/Chars.java

```diff
@@ -568,8 +557,8 @@ public Chars next() {
                 } else {
                     // lf or cr (exactly one is != -1 and max returns that one)
                     int idx = Math.max(cr, lf);
-                    resetLookahead(cr, idx);
                     pos = idx + 1;
+                    resetLookahead(cr, idx);
                     return subSequence(curPos, idx);
                 }
             }
```

Comment:
Pros:

1, better code readability:
2, Reduced complexity: This modification reduces the number of branching procedures to handle different cases, thus reducing the complexity of the code.

Disadvantages:

May reduce efficiency or may not be able to handle other cases: The need to adopt this change must be judged against the overall project requirements.